### PR TITLE
Google Sign In Integration

### PR DESCRIPTION
**Pull request by vineet-webonise**

### CODE REVIEW

File: SUNFLOWR/SUNFLOWR/AppDelegate.swift [hunk 1]
```diff
@@ -31,6 +35,33 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
     }
 
+    @available(iOS 9.0, *)
+    func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool {
+      return GIDSignIn.sharedInstance().handle(url)
+    }
+
```
Comment (saketPandhare-Webonise) [inline SUNFLOWR/SUNFLOWR/AppDelegate.swift:42]: Is this method used when user signs up by google ?

Comment (vineet-webonise) [reply SUNFLOWR/SUNFLOWR/AppDelegate.swift:42]: handle the URL that application receives at the end of the authentication process

File: SUNFLOWR/SUNFLOWR/AppDelegate.swift [hunk 2]
```diff
@@ -31,6 +35,33 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+    @available(iOS 9.0, *)
+    func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool {
+      return GIDSignIn.sharedInstance().handle(url)
+    }
+    
+    // MARK: - Google Login Delegate Methods
+    func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!,
+              withError error: Error!) {
+      if let error = error {
+        if (error as NSError).code == GIDSignInErrorCode.hasNoAuthInKeychain.rawValue {
```
Comment (saketPandhare-Webonise) [inline SUNFLOWR/SUNFLOWR/AppDelegate.swift:47]: Hey @vineet-webonise  i was asking of above function ,sorry for commenting on wrong method 

`func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!,
               withError error: Error!)`

can you let me know when this method is called

Comment (vineet-webonise) [reply SUNFLOWR/SUNFLOWR/AppDelegate.swift:47]: Its GIDSignInDelegate protocol to handle the sign-in process

Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/AppDelegate.swift:47]: As this is SignIn Delegate move this method to LoginController or SignUpController. Is it really required to be in AppDelegate ?

Comment (vineet-webonise) [reply SUNFLOWR/SUNFLOWR/AppDelegate.swift]: Done

**Pull request merged**